packages/monaco/src/browser/monaco-quick-open-service.ts

```diff
@@ -346,12 +346,23 @@ export class MonacoQuickOpenControllerOptsImpl implements MonacoQuickOpenControl
         this.options.onClose(cancelled);
     }
 
-    private toOpenModel(lookFor: string, items: QuickOpenItem[], actionProvider?: QuickOpenActionProvider): monaco.quickOpen.QuickOpenModel {
+    protected toOpenModel(lookFor: string, items: QuickOpenItem[], actionProvider?: QuickOpenActionProvider): monaco.quickOpen.QuickOpenModel {
         const entries: monaco.quickOpen.QuickOpenEntry[] = [];
-        for (const item of items) {
+        let groupOptionsToMerge: QuickOpenGroupItemOptions | undefined = undefined;
+        for (let item of items) {
+            if (!(item instanceof QuickOpenGroupItem) && groupOptionsToMerge) {
+                item = new QuickOpenGroupItem({ ...groupOptionsToMerge, ...item.options });
```

Comment:
what if `item` is a subclass of `QuickOpenItem`? Maybe we have `QuickOpenGroupItem` which is implemented by delegating to `QuickOpenItem`?

Reply:
I do not understand it. Do you have an example in mind?

Reply:
Look for `extends QuickOpenItem`, for instance here: https://github.com/eclipse-theia/theia/blob/db8c80783655a25629aace1e5d5dd5f544f086d3/packages/scm/src/browser/scm-quick-open-service.ts#L73-L100